tests/Unit/Evolution/DiscontinuousGalerkin/Test_InboxTags.cpp

```diff
@@ -0,0 +1,180 @@
+  inbox.erase(time_step_id_b);
+  CHECK(inbox.count(time_step_id_b) == 0);
+}
+
+template <size_t Dim>
+void test_with_ghost_cells() {
+  static constexpr size_t number_of_components = 1 + Dim;
+  using bc_tag = Tags::BoundaryCorrectionAndGhostCellsInbox<Dim>;
+  using Type = std::tuple<Mesh<Dim - 1>, std::optional<std::vector<double>>,
+                          std::optional<std::vector<double>>, ::TimeStepId>;
+  using Inbox = typename bc_tag::type;
+
+  std::uniform_real_distribution<double> dist(-1.0, 2.3);
+  MAKE_GENERATOR(gen);
+
+  const TimeStepId time_step_id_a{true, 3, Time{Slab{0.2, 3.4}, {3, 100}}};
+  const TimeStepId time_step_id_b{true, 4, Time{Slab{3.4, 5.4}, {13, 100}}};
+  const TimeStepId time_step_id_c{true, 5, Time{Slab{5.4, 6.4}, {17, 100}}};
+  const std::pair nhbr_key{Direction<Dim>::lower_xi(), ElementId<Dim>{1}};
+
+  Inbox inbox{};
+
+  // Send ghost cells first
+  Type send_data_a{};
+  get<0>(send_data_a) = Mesh<Dim - 1>{5, Spectral::Basis::Legendre,
+                                      Spectral::Quadrature::GaussLobatto};
+  get<1>(send_data_a) = std::vector<double>(
+      get<0>(send_data_a).number_of_grid_points() * number_of_components);
+  get<3>(send_data_a) = time_step_id_a;
+  fill_with_random_values(make_not_null(&*get<1>(send_data_a)),
+                          make_not_null(&gen), make_not_null(&dist));
+
+  bc_tag::insert_into_inbox(make_not_null(&inbox), time_step_id_a,
+                            std::make_pair(nhbr_key, send_data_a));
+
+  CHECK((inbox.at(time_step_id_a).at(nhbr_key) == send_data_a));
+
+  Type send_data_b{};
+  get<0>(send_data_b) = Mesh<Dim - 1>{7, Spectral::Basis::Legendre,
+                                      Spectral::Quadrature::GaussLobatto};
+  get<1>(send_data_b) = std::vector<double>(
+      get<0>(send_data_b).number_of_grid_points() * number_of_components);
+  get<3>(send_data_b) = time_step_id_b;
+  fill_with_random_values(make_not_null(&*get<1>(send_data_b)),
+                          make_not_null(&gen), make_not_null(&dist));
+
+  bc_tag::insert_into_inbox(make_not_null(&inbox), time_step_id_b,
+                            std::make_pair(nhbr_key, send_data_b));
+
+  CHECK((inbox.at(time_step_id_a).at(nhbr_key) == send_data_a));
+  CHECK((inbox.at(time_step_id_b).at(nhbr_key) == send_data_b));
+
+  inbox.erase(time_step_id_a);
+  CHECK(inbox.count(time_step_id_a) == 0);
+
+  CHECK((inbox.at(time_step_id_b).at(nhbr_key) == send_data_b));
+  inbox.erase(time_step_id_b);
+  CHECK(inbox.count(time_step_id_b) == 0);
+
+  // Now send ghost cells and fluxes separately. Ghost cells first!
+  bc_tag::insert_into_inbox(make_not_null(&inbox), time_step_id_a,
+                            std::make_pair(nhbr_key, send_data_a));
+  Type send_flux_data_a;
+  get<0>(send_flux_data_a) = get<0>(send_data_a);
+  get<2>(send_flux_data_a) = std::vector<double>(
+      get<0>(send_data_a).number_of_grid_points() * number_of_components);
+  get<3>(send_flux_data_a) = time_step_id_c;
```

Comment:
Here it might be helpful to add a short `// check that TimeStepId is updated with value from second send` or similar?